src/DataProtection/Abstractions/src/IDataProtector.cs

```diff
@@ -24,5 +27,34 @@ public interface IDataProtector : IDataProtectionProvider
         /// Thrown if the protected data is invalid or malformed.
         /// </exception>
         byte[] Unprotect(byte[] protectedData);
+
+
+#if NETCOREAPP
+        /// <summary>
+        /// Cryptographically protects a piece of plaintext data.
+        /// </summary>
+        /// <param name="output">Where to write the protected form of the plaintext data.</param>
+        /// <param name="plaintext">The plaintext data to protect.</param>
+        /// <param name="bytesWritten">The number of bytes written to output.</param>
+        /// <returns>true if output is long enough to receive the protected data; otherwise, false.</returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0016:Add public types and members to the declared API", Justification = "<Pending>")]
+        bool TryProtect(Span<byte> output, ReadOnlySpan<byte> plaintext, out int bytesWritten)
```

Comment:
Nit: Might want to reverse the order of inputs and outputs:

```suggestion
        bool TryProtect(ReadOnlySpan<byte> plaintext, Span<byte> output, out int bytesWritten)
```